src/test/FuzzerImpl.cpp

```diff
@@ -447,31 +651,49 @@ class FuzzTransactionFrame : public TransactionFrame
                          TransactionEnvelope const& envelope)
         : TransactionFrame(networkID, envelope){};
 
-    void
-    attemptApplication(Application& app, AbstractLedgerTxn& ltx)
+    bool
+    checkValidNoSignature(AbstractLedgerTxn& ltx)
     {
         // reset results of operations
         resetResults(ltx.getHeader(), 0, true);
 
-        // attempt application of transaction without accounting for sequence
-        // number, processing the fee, or committing the LedgerTxn
         SignatureChecker signatureChecker{
             ltx.loadHeader().current().ledgerVersion, getContentsHash(),
             mEnvelope.v1().signatures};
         // if any ill-formed Operations, do not attempt transaction application
         auto isInvalidOperationXDR = [&](auto const& op) {
-            return !op->checkValid(signatureChecker, ltx, false);
+            auto const notValid = !op->checkValid(signatureChecker, ltx, false);
+            LOG_TRACE(DEFAULT_LOG, "{} operation {}",
+                      notValid ? "INvalid" : "VALID",
+                      xdr_to_string(op->getOperation()));
+            return notValid;
         };
-        if (std::any_of(mOperations.begin(), mOperations.end(),
-                        isInvalidOperationXDR))
+        bool const valid = !std::any_of(mOperations.begin(), mOperations.end(),
+                                        isInvalidOperationXDR);
+        LOG_TRACE(DEFAULT_LOG, "{} transaction {}", valid ? "VALID" : "INvalid",
+                  xdr_to_string(this->getEnvelope()));
+        return valid;
+    }
+
+    // attempt application of transaction without accounting for sequence
+    // number, processing the fee, or committing the LedgerTxn
+    void
+    attemptApplication(Application& app, AbstractLedgerTxn& ltx)
+    {
+        if (!checkValidNoSignature(ltx))
         {
+            markResultFailed();
             return;
         }
         // while the following method's result is not captured, regardless, for
         // protocols < 8, this triggered buggy caching, and potentially may do
         // so in the future
         loadSourceAccount(ltx, ltx.loadHeader());
+        processSeqNum(ltx);
         TransactionMeta tm(2);
+        SignatureChecker signatureChecker{
```

Comment:
why do we need to instantiate 2 signature checkers?

Reply:
I made the checker a member of `Fuzzer` so that it can be reused by `checkValidNoSignature()` when called from `attemptApplication()` (and is also available for `checkValidNoSignature()` when called directly by `genFuzz()`).